pkg/distributor/distributor_test.go

```diff
@@ -288,8 +288,18 @@ func TestDistributor_Push(t *testing.T) {
 
 			request := makeWriteRequest(tc.samples.startTimestampMs, tc.samples.num, tc.metadata, false)
 			response, err := ds[0].Push(ctx, request)
-			assert.Equal(t, tc.expectedResponse, response)
-			assert.Equal(t, tc.expectedError, err)
+
+			if tc.expectedError == nil {
+				assert.Equal(t, emptyResponse, response)
+				assert.Nil(t, err)
```

Comment:
It's best to halt the test immediately on an unexpected error IMO:
```suggestion
				require.NoError(t, err)
				assert.Equal(t, emptyResponse, response)
```